src/audio/google_audio_post_processing_mock.c

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Google LLC.
+//
+// Author: Kehuang Li <kehuangli@google.com>
+
+#include "google_audio_post_processing.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "sof/lib/alloc.h"
+#include "ipc/topology.h"
+
+#define GOOGLE_AUDIO_POST_PROCESSING_SAMPLE_RATE_HZ 48000
+
+struct GoogleAudioPostProcessingState {
+	uint8_t config[8];
+	int num_channels;
+	int num_frames;
+	int volume[SOF_IPC_MAX_CHANNELS];
+};
+
+GoogleAudioPostProcessingState *GoogleAudioPostProcessingCreate(void)
+{
+	struct GoogleAudioPostProcessingState *s = rballoc(
+		0, SOF_MEM_CAPS_RAM, sizeof(GoogleAudioPostProcessingState));
+
+	s->num_channels = 1;
+	s->num_frames = 0;
+	return s;
+}
+
+void GoogleAudioPostProcessingDelete(GoogleAudioPostProcessingState *state)
+{
+	if (state != NULL)
+		rfree(state);
+}
+
+int GoogleAudioPostProcessingSetup(GoogleAudioPostProcessingState *const state,
+				   int channels, int frames, int volume,
+				   const uint8_t *const config, int config_size)
+{
+	if (config_size > sizeof(state->config))
+		return -ENOMEM;
+
+	memcpy_s(state->config, sizeof(state->config), config, config_size);
```

Comment:
why not return the return value or memcpy_s?

Reply:
Done.